python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -985,6 +987,32 @@ def fetch_row_counts(
                 A set of corresponding Dagster events for dbt models, with row counts attached,
                 yielded in the order they are emitted by dbt.
         """
+        return self._attach_metadata(_fetch_row_count_metadata)
+
+    def _attach_metadata(
+        self,
+        fn: Callable[[DbtCliInvocation, DbtDagsterEventType], Optional[Dict[str, Any]]],
+    ) -> "DbtEventIterator[DbtDagsterEventType]":
+        """Runs a threaded task to attach metadata to each event in the iterator.
+
+        Args:
+            fn (Callable[[DbtCliInvocation, DbtDagsterEventType], Optional[Dict[str, Any]]]):
```

Comment:
I think including `DbtCliInvocation` is unnecessary.

- We can access it from `self`, if we make `_dbt_cli_invocation` public.
- From the user's perspective, `DbtCliInvocation` will be accessible from the scope outside of the closure, so it's repetitive.

Reply:
Got it, I missed that. Makes sense.